### PR TITLE
Feature/icmpv6 ra

### PR DESCRIPTION
**Pull request by msvoelker**
Enables packetdrill to inject ICMPv6 Router Advertisements with the MTU option.

### CODE REVIEW

File: gtests/net/packetdrill/config.c [hunk 1]
```diff
@@ -79,7 +81,9 @@ struct option options[] = {
 	{ "connect_port",	.has_arg = true,  NULL, OPT_CONNECT_PORT },
 	{ "remote_ip",		.has_arg = true,  NULL, OPT_REMOTE_IP },
 	{ "local_ip",		.has_arg = true,  NULL, OPT_LOCAL_IP },
+	{ "local_linklocal_ip", .has_arg = true, NULL, OPT_LOCAL_LINKLOCAL_IP },
 	{ "gateway_ip",		.has_arg = true,  NULL, OPT_GATEWAY_IP },
+	{ "gateway_linklocal_ip", .has_arg = true, NULL, OPT_GATEWAY_LINKLOCAL_IP },
```
Comment (tuexen) [inline gtests/net/packetdrill/config.c:86]: Please reindent the other lines to get thinks aligned again.

**Review by tuexen — COMMENTED** [review]
I guess you testing it on FreeBSD, did you also test on Linux and or MacOS?

File: gtests/net/packetdrill/config.c (hunk 1 above)
Comment (msvoelker) [reply gtests/net/packetdrill/config.c]: Done.

File: gtests/net/packetdrill/config.c [hunk 2]
```diff
@@ -176,7 +182,9 @@ void show_usage(void)
 
 #define DEFAULT_V6_LIVE_REMOTE_IP_STRING   "2001:DB8::1/32"
 #define DEFAULT_V6_LIVE_LOCAL_IP_STRING    "fd3d:fa7b:d17d::1"
+#define DEFAULT_V6_LIVE_LOCAL_LINKLOCAL_IP_STRING    "fe80::1"
 #define DEFAULT_V6_LIVE_GATEWAY_IP_STRING  "fd3d:fa7b:d17d::2"
+#define DEFAULT_V6_LIVE_GATEWAY_LINKLOCAL_IP_STRING  "fe80::2"
```
[the remark this answers is not in the document]
Comment (msvoelker) [reply gtests/net/packetdrill/config.c]: Done.

File: gtests/net/packetdrill/config.c [hunk 3]
```diff
@@ -238,7 +252,9 @@ void set_default_config(struct config *config)
 
 	config->live_remote_ip_string[0]	= '\0';
 	config->live_local_ip_string[0]		= '\0';
+	config->live_local_linklocal_ip_string[0] = '\0';
 	config->live_gateway_ip_string[0]	= '\0';
+	config->live_gateway_linklocal_ip_string[0] = '\0';
```
[the remark this answers is not in the document]
Comment (msvoelker) [reply gtests/net/packetdrill/config.c:257]: Done.

File: gtests/net/packetdrill/icmp_packet.c [hunk 4]
```diff
@@ -120,6 +120,7 @@ struct icmp_type_info icmpv6_types[] = {
 	{ ICMPV6_PARAMPROB,	"parameter_problem", icmpv6_paramprob_codes },
 	{ ICMPV6_ECHO_REQUEST,  "echo_request" },
 	{ ICMPV6_ECHO_REPLY,    "echo_reply" },
+	{ ICMPV6_ROUTER_ADVERTISEMENT, "router_advertisement" },
```
[the remark this answers is not in the document]
Comment (msvoelker) [reply gtests/net/packetdrill/icmp_packet.c:123]: Done.

File: gtests/net/packetdrill/icmp_packet.c [hunk 5]
```diff
@@ -182,18 +183,31 @@ static int set_icmpv6_header(struct icmpv6 *icmpv6, u8 type, u8 code,
...
 			return STATUS_ERR;
 		}
 		if (!is_valid_u32(mtu)) {
 			asprintf(error, "ICMPv6 MTU out of 32-bit range");
 			return STATUS_ERR;
 		}
-		icmpv6->message.packet_too_big.mtu = htonl(mtu);
+		if (type == ICMPV6_PKT_TOOBIG) {
+			icmpv6->message.packet_too_big.mtu = htonl(mtu);
+		} else { // if (type == ICMPV6_ROUTER_ADVERTISEMENT)
```
[the remark this answers is not in the document]
Comment (msvoelker) [reply gtests/net/packetdrill/icmp_packet.c]: Done.

**Comment by msvoelker** [comment]
> I guess you testing it on FreeBSD, did you also test on Linux and or MacOS?

I tested on Linux and after reading your comment on MacOS. I changed the code to keep it MacOS compatible. For both OS, I did not check whether they process the Router Advertisement.

**Pull request merged**